aderyn_core/src/report/mod.rs

```diff
@@ -109,7 +109,7 @@ impl ContextLoader {
     pub fn files_summary(&self) -> FilesSummary {
         FilesSummary {
             total_source_units: self.source_units.len(),
```

Comment:
@alexroan  please can you clarify that `source_unitls.len()` will be equal to `src_filepaths.len()` that you newly added ?

Reply:
If yes, then I think that we need to somehow enforce that equality in code by making another abstraction somewhere coz I am generally a fan of single source of truth... But I also understand that it's not always feasible :P

Reply:
Great question...

I learnt yesterday that `source_units.len()` != `src_filepaths.len()`.

Tl;dr:
* `source_units.len()` = the total number of AST source units loaded (there may be duplicates based on solc version)
* `src_filepaths.len()` = the total number of solidity files.

Why?:

When Foundry compiles the source files, sometimes Foundry compiles and outputs to multiple files, like this:
![image](https://github.com/Cyfrin/aderyn/assets/6523673/b3f1e926-4d96-4826-8728-e7d216eabc18)

This occurs when more than one specific solidity pragma version is defined in contracts, and there are also contracts with wide pragmas.

Aderyn loads both ASTs, even though they represent the same source file. We _have_ to load both because of the way the AST objects within them reference each other.

Reply:
✌️